### PR TITLE
tests/e2e: isolate framework from tests, restructure files

### PR DESCRIPTION
**Pull request by eduser25**
New file structure:

- tests/
  - framework/
    -  Contains framework files. Importing these will automatically
       set up init, BeaforeEach, AfterEach... hooks we usually run
        between tests.
  - e2e/
    - contains end-to-end tests.
  - scenarios/
    - Contains larger, use-case driven tests that might benefit from
      using framework itself.

Changes:
- Updated some README.md references
- Moved framework specific files to "framework" folder.
- Made most of the usable framework structures and attributess
public for importing modules/tests to use.
- Moved mutating_webhook scenario under scenarios.

**Affected area**:

- Tests                  [X]
- CI System              [X]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
No

### CODE REVIEW

File: tests/framework/common.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package e2e
+package e2ef
```
Comment (nojnhuh) [inline tests/framework/common.go:1]: Any particular reason this isn't `package framework` to match the directory name?

**Review by nojnhuh — APPROVED** [review]
Assuming the CI failure is a flake, LGTM.

File: tests/framework/common.go (hunk 1 above)
Comment (eduser25) [reply tests/framework/common.go:1]: It was initially like that... and changed it. Can't even reason why.
I'll fix.

**Review by snehachhabria — APPROVED** [review]

**Review by nojnhuh — APPROVED** [review]

**Review by shashankram — APPROVED** [review]

File: tests/scenarios/suite_test.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package scenarios
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func TestSuite(t *testing.T) {
```
Comment (draychev) [inline tests/scenarios/suite_test.go:10]: Thanks!

**Review by draychev — APPROVED** [review]

**Pull request merged**